yt_dlp/extractor/bilibili.py

```diff
@@ -155,7 +171,53 @@ def _get_episodes_from_season(self, ss_id, url):
         for entry in traverse_obj(season_info, (
                 'result', 'main_section', 'episodes',
                 lambda _, v: url_or_none(v['share_url']) and v['id'])):
-            yield self.url_result(entry['share_url'], BiliBiliBangumiIE, f'ep{entry["id"]}')
+            yield self.url_result(entry['share_url'], BiliBiliBangumiIE, str_or_none(entry.get('id')))
+
+    def _get_divisions(self, video_id, graph_version, edges, edge_id, cid_edges={}):
+        division_data = self._download_json(
+            'https://api.bilibili.com/x/stein/edgeinfo_v2', video_id,
```

Comment:
could you provide a sample of what the response from this endpoint looks like?

Reply:
For a video like https://www.bilibili.com/video/BV1af4y1H7ga/ (also added as a test case), the `play_info` dict will have all necessay to initiate a query of selections/paths. Browser can then render the choices/following "edges" on screen, and query the the selected "edge", until user reaches an ending, or just loop infinitely.

Here are some example queries and responses

`story_list` keeps (by server) the list of "edges" that user has visited, so that browser can render an array of previous choices to allow user to jump back to a previous choice. The `story_list` contains only the first "edge" when no cookies is provided. However, the first "edge" is always included. Traversing `story_list` ensures that we can always have a way to get the info of the beginning first "edge", which does not appear in any choices.

The `questions -> choices` has all choices that user can select in the current video. They have the id for the "edge" and also `cid` for the video that the edge may use. Different edge may reuse the same video, i.e., has the same `cid`, which means we can download the same video multiple times if we use the id of an "edge" as primary key.

`cover` here are just snapshots from a video frame, no need for extraction.

There are also variables, allowing creating conditions for different choices. They do not matter since we ignore conditions (if any) on choices while we traverse.

```
https://api.bilibili.com/x/stein/edgeinfo_v2?graph_version=570522&choices=&aid=292734508

{
   "code":0,
   "message":"0",
   "ttl":1,
   "data":{
      "title":"听见猫猫叫~",
      "edge_id":1,
      "story_list":[
         {
            "node_id":1,
            "edge_id":1,
            "title":"听见猫猫叫~",
            "cid":400950101,
            "start_pos":10000,
            "cover":"http://i0.hdslb.com/bfs/steins-gate/400950101_screenshot.jpg",
            "is_current":1,
            "cursor":0
         }
      ],
      "edges":{
         "dimension":{
            "width":1920,
            "height":1080,
            "rotate":0,
            "sar":""
         },
         "questions":[
            {
               "id":0,
               "type":2,
               "start_time_r":300,
               "duration":-1,
               "pause_video":1,
               "title":"",
               "choices":[
                  {
                     "id":20420805,
                     "platform_action":"JUMP 20420805 400958579",
                     "native_action":"",
                     "condition":"",
                     "cid":400958579,
                     "x":1056,
                     "y":749,
                     "text_align":2,
                     "option":"A 回家打游戏",
                     "is_default":1
                  },
                  {
                     "id":20420806,
                     "platform_action":"JUMP 20420806 400951416",
                     "native_action":"$DjDV_64_GqZP=$DjDV_64_GqZP+30.00",
                     "condition":"",
                     "cid":400951416,
                     "x":387,
                     "y":668,
                     "text_align":2,
                     "option":"B 去看看猫猫"
                  },
                  {
                     "id":20420807,
                     "platform_action":"JUMP 20420807 400950118",
                     "native_action":"",
                     "condition":"",
                     "cid":400950118,
                     "x":607,
                     "y":252,
                     "text_align":2,
                     "option":"C 直接给我自我介绍！！！"
                  }
               ]
            }
         ],
         "skin":{
            "choice_image":"http://i0.hdslb.com/bfs/app/62fb3ea57a93a796ed2b371e583720161c8e97ba.png",
            "title_text_color":"000000ff",
            "title_shadow_color":"00000000",
            "progressbar_color":"ffffffff",
            "progressbar_shadow_color":"000000cc"
         }
      },
      "buvid":"0A0E2BAD-C443-5F02-AFFD-8B7ADC0EC56E36757infoc",
      "preload":{
         "video":[
            {
               "aid":292734508,
               "cid":400958579
            },
            {
               "aid":292734508,
               "cid":400951416
            },
            {
               "aid":292734508,
               "cid":400950118
            }
         ]
      },
      "hidden_vars":[
         {
            "value":0,
            "id":"v-8#EnfmGs#e",
            "id_v2":"$8_35_EnfmGs_35_e",
            "type":1,
            "is_show":0,
            "name":"数值4",
            "skip_overwrite":0
         },
         {
            "value":0,
            "id":"v-DjDV@GqZP",
            "id_v2":"$DjDV_64_GqZP",
            "type":1,
            "is_show":0,
            "name":"数值1",
            "skip_overwrite":0
         },
         {
            "value":0,
            "id":"v-d3B2Vh#brm",
            "id_v2":"$d3B2Vh_35_brm",
            "type":1,
            "is_show":0,
            "name":"数值2",
            "skip_overwrite":0
         },
         {
            "value":0,
            "id":"v-gKgiX4THkq",
            "id_v2":"$gKgiX4THkq",
            "type":1,
            "is_show":0,
            "name":"数值3",
            "skip_overwrite":0
         },
         {
            "value":98,
            "id":"v-rH2CKB0SDO",
            "id_v2":"$rH2CKB0SDO",
            "type":2,
            "is_show":0,
            "name":"随机值",
            "skip_overwrite":0
         }
      ],
      "is_leaf":0
   }
}
```

Here if I click the first choice "A 回家打游戏" ("id":20420805), the next query would be like this (omitting irrelevent part):
note if we have cookies, we will have a `story_list` containing more than the beginning "edge" here.

```
https://api.bilibili.com/x/stein/edgeinfo_v2?graph_version=570522&edge_id=20420805&aid=292734508

{
   ...
   "data":{
      "title":"吸入异世界",
      "edge_id":20420805,
      "story_list":[
         {
            "node_id":1,
            "edge_id":1,
            "title":"听见猫猫叫~",
            "cid":400950101,
            "start_pos":10000,
            "cover":"http://i0.hdslb.com/bfs/steins-gate/400950101_screenshot.jpg",
            "cursor":0
         },
         {
            "node_id":20420805,
            "edge_id":20420805,
            "title":"吸入异世界",
            "cid":400958579,
            "start_pos":36000,
            "cover":"http://i0.hdslb.com/bfs/steins-gate/400958579_screenshot.jpg",
            "is_current":1,
            "cursor":1
         }
      ],
      "edges":{
         "dimension":{
            "width":1920,
            "height":1080,
            "rotate":0,
            "sar":""
         },
         "questions":[
            {
               "id":0,
               "type":1,
               "start_time_r":300,
               "duration":-1,
               "pause_video":1,
               "title":"",
               "choices":[
                  {
                     "id":20420829,
                     "platform_action":"JUMP 20420829 400953418",
                     "native_action":"",
                     "condition":"",
                     "cid":400953418,
                     "option":"A 奶妈？我不玩",
                     "is_default":1
                  },
                  {
                     "id":20420830,
                     "platform_action":"JUMP 20420830 400953719",
                     "native_action":"",
                     "condition":"",
                     "cid":400953719,
                     "option":"B 我在哪？你是谁？"
                  },
                  {
                     "id":20420831,
                     "platform_action":"JUMP 20420831 400954186",
                     "native_action":"",
                     "condition":"",
                     "cid":400954186,
                     "option":"C 好可爱！prprpr"
                  }
               ]
            }
         ],
         "skin":{ ... }
      },
      "preload":{
         "video":[ ... ]
      },
      "hidden_vars":[ ... ],
      "is_leaf":0
   }
}
```